vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/NetworkActions.java

```diff
@@ -0,0 +1,190 @@
+import io.appium.java_client.android.connection.ConnectionStateBuilder;
+
+public class NetworkActions
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworkActions.class);
+
+    private static final String IOS_PREFERENCES_BUNDLE_ID = "com.apple.Preferences";
+
+    private final IWebDriverProvider webDriverProvider;
+    private final GenericWebDriverManager genericWebDriverManager;
+    private final ApplicationActions applicationActions;
+    private final IBaseValidations baseValidations;
+
+    public NetworkActions(IWebDriverProvider webDriverProvider, GenericWebDriverManager genericWebDriverManager,
+            ApplicationActions applicationActions, IBaseValidations baseValidations)
+    {
+        this.webDriverProvider = webDriverProvider;
+        this.genericWebDriverManager = genericWebDriverManager;
+        this.applicationActions = applicationActions;
+        this.baseValidations = baseValidations;
+    }
+
+    public void changeNetworkConnectionState(NetworkToggle networkToggle, Mode mode)
+    {
+        if (genericWebDriverManager.isAndroid())
+        {
+            ConnectionStateBuilder connectionStateBuilder = NetworkToggle.ON == networkToggle
+                    ? mode.getEnablingConnectionStateBuilder()
+                    : mode.getDisablingConnectionStateBuilder();
+            webDriverProvider.getUnwrapped(AndroidDriver.class).setConnection(connectionStateBuilder.build());
+        }
+        if (genericWebDriverManager.isIOS())
+        {
+            changeNetworkConnectionStateForIOS(networkToggle, mode);
+        }
+    }
+
+    private void changeNetworkConnectionStateForIOS(NetworkToggle networkToggle, Mode mode)
+    {
+        applicationActions.activateApp(IOS_PREFERENCES_BUNDLE_ID);
+        findMenuItemInSettings(new Locator(AppiumLocatorType.ACCESSIBILITY_ID, mode.getIOSNetworkConnectionAlias()))
+                .ifPresentOrElse(item ->
+                {
+                    item.click();
+                    findMenuItemInSettings(new Locator(AppiumLocatorType.IOS_CLASS_CHAIN, String
+                            .format("**/XCUIElementTypeSwitch[`label == '%s'`]", mode.getIOSNetworkConnectionAlias())))
+                            .ifPresent(switchButton ->
+                            {
+                                if (!networkToggle.active(switchButton.getAttribute("value")))
+                                {
+                                    switchButton.click();
+                                }
+                                else
+                                {
+                                    LOGGER.atInfo().addArgument(mode).addArgument(networkToggle)
+                                            .log("{} is already {}.");
+                                }
+                            });
+                }, () -> LOGGER.atWarn().addArgument(mode.getIOSNetworkConnectionAlias())
+                        .log("{} is not presented in iOS Preferences."));
```

Comment:
```suggestion
        findMenuItemInSettings(new Locator(AppiumLocatorType.ACCESSIBILITY_ID, mode.getIOSNetworkConnectionAlias()))
                .ifPresent(item ->
                {
                    item.click();
                    findMenuItemInSettings(new Locator(AppiumLocatorType.IOS_CLASS_CHAIN, String
                            .format("**/XCUIElementTypeSwitch[`label == '%s'`]", mode.getIOSNetworkConnectionAlias())))
                            .ifPresent(switchButton ->
                            {
                                if (!networkToggle.active(switchButton.getAttribute("value")))
                                {
                                    switchButton.click();
                                }
                                else
                                {
                                    LOGGER.atInfo().addArgument(mode).addArgument(networkToggle)
                                            .log("{} is already {}.");
                                }
                            });
                });
```
base validations will record the failure in case of missing element